### ISSUE TITLE
Client side bundle includes react-dom twice

### ISSUE THREAD

**Issue by fabio-nettis** [vercel/next.js#51508]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.5.0: Mon Apr 24 20:51:50 PDT 2023; root:xnu-8796.121.2~5/RELEASE_X86_64
    Binaries:
      Node: 20.3.0
      npm: 9.6.7
      Yarn: 3.5.0
      pnpm: N/A
    Relevant packages:
      next: 13.4.6
      eslint-config-next: 13.4.6
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue or a replay of the bug

![react-dom-issue](https://github.com/vercel/next.js/assets/36916632/ebeeed14-a76f-4757-b872-649db59b7120)

### To Reproduce

1. Create a new next project using `yarn create next-app test`
2. Go through installation process
3. Add the bundle analyzer
4. Analyze the bundle

### Describe the Bug

Using the analyzer to check my bundle size, and I've noticed that react-dom is included twice. Where one version is the compiled version that is included in next.js itself and the other is the one installed in the project.

### Expected Behavior

 One instance should be enough, since it is one of the chunkiest parts of my bundle.

### Which browser are you using? (if relevant)

Chrome 114.0.5735.106 (Official Build) (x86_64)

### How are you deploying your application? (if relevant)

next start

**Comment by huozhi** [comment]
The graph contains all the possible to load chunks for the whole app, not per page. If you use the filter on the left it can help you filter the entry (the page and the related chunks) to load

![image](https://github.com/vercel/next.js/assets/4800338/0937873c-b043-4b19-adfe-9f3d1c6f668b)


Framework chunk is for pages, so it only exists in pages for instance default _error page, when you're rendering pages under `app/ ` it won't be included

**Comment by fabio-nettis** [comment]
@huozhi Interesting you should say that, because excluding one of the both via import modularization in combination with a custom webpack config cuts the client bundle size in half.

##### Next.js config

```js
const nextConfig = {
  // redacted
  webpack: (config, { isServer, dev }) => {
    config.module.rules.push({
      test: /react-dom/,
      loader: "null-loader",
    });

    return config;
  },
  modularizeImports: {
    "react-dom/([A-Z,a-z]*)": {
      transform: "next/dist/compiled/react-dom/{{ matches.[1] }}",
      skipDefaultConversion: true,
    },
  },
};
```

Sadly this approach some stuff and I am wondering how they correlate.

**Comment by huozhi** [comment]
There're configured in separate layers, I think this won't easily change the internal configured webpack rules. Can you share what are you trying to do here? Are you trying to use the same version of react for pages and why?

**Comment by fabio-nettis** [comment]
I am currently optimizing the bundle size for a quite large e-commerce application that is being migrated from pages to the app directory, now when analyzing the bundle I have noticed the client bundle includes react-dom twice (as seen in the image above).  As you mentioned this would be for the pages directory.

> Framework chunk is for pages, so it only exists in pages for instance default _error page, when you're rendering pages under app/  it won't be included.

This is absolutely clear to me, but not only do we not use the pages directory anymore, but there is not even a pages folder anymore. To me this seems like a error to be bundled at all, so I have tried to exclude on of the two versions. If I try to remove the dependency completely and switch it out with the compiled version already present in the bundle the size for all pages inside the app directory is cut in half, from 88kb to around ~44kb. Thats also the case if I try a reversed webpack configuration.

```js
const nextConfig = {
  // redacted
  webpack: (config, { isServer, dev }) => {
    config.module.rules.push({
      test: /next\/dist\/compiled\/react-dom/,
      loader: "null-loader",
    });

    return config;
  },
  modularizeImports: {
    "next/dist/compiled/react-dom(/?[A-Z,a-z]*)": {
      transform: "react-dom{{ matches.[1] }}",
      skipDefaultConversion: true,
    },
  },
};
```

The app still renders ok, is blazingly fast, but things like dynamic imports do not work anymore and the error in the console states: `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.Dispatcher` is undefined, to me it seems like one of the two versions definitely should not be there.
